target/ppc/translate.c

```diff
@@ -7457,6 +7457,15 @@ static int times_4(DisasContext *ctx, int x)
 # define REQUIRE_64BIT(CTX)  REQUIRE_INSNS_FLAGS(CTX, 64B)
 #endif
 
+#define VOID
+#define REQUIRE_VECTOR(CTX, RET)                        \
```

Comment:
I think a variadic macro can be used here, so you don't need `#define VOID` and can just use `REQUIRE_VECTOR(ctx)` in legacy decode methods. E.g.:
```c
#define REQUIRE_VECTOR(CTX, ...)                        \
    do {                                                \
        if (unlikely(!(CTX)->altivec_enabled)) {        \
            gen_exception((CTX), POWERPC_EXCP_VPU);     \
            return __VA_ARGS__;                         \
        }                                               \
    } while (0)
```

Reply:
idk, because Luis was just saying how we shouldn't have an extra argument for decode tree, since everything will be moved to it later. maybe we can just ignore the legacy stuff, have it always return true as it did before and other instructions will use it as we move them to decodetree.

Reply:
In that case I think we should keep it in `vector-impl,inc.c`, since it wouldn't be used anywhere else (at least for now).